File: docs/snippets/all/views/state_timeline.py
# Use a blueprint to show a StateTimelineView.

import rerun as rr
import rerun.blueprint as rrb

rr.init("rerun_example_state_timeline", spawn=True)

rr.set_time("step", sequence=0)
rr.log("door", rr.StateChange(state="open"))

rr.set_time("step", sequence=1)
rr.log("door", rr.StateChange(state="closed"))

rr.set_time("step", sequence=2)
rr.log("door", rr.StateChange(state="open"))

# Create a state timeline view to display the state transitions.
blueprint = rrb.Blueprint(
    rrb.StateTimelineView(
        origin="/",
        name="State Transitions",
    ),
    collapse_panels=True,
)

rr.send_blueprint(blueprint)

File: docs/snippets/all/views/state_timeline.rs
//! Use a blueprint to show a StateTimelineView.

use rerun::blueprint::{Blueprint, StateTimelineView};

fn main() -> Result<(), Box<dyn std::error::Error>> {
    let blueprint = Blueprint::new(
        StateTimelineView::new("State Transitions").with_origin("/"),
    );

    let rec =
        rerun::RecordingStreamBuilder::new("rerun_example_state_timeline")
            .with_blueprint(blueprint)
            .spawn()?;

    rec.set_time_sequence("step", 0);
    rec.log("door", &rerun::StateChange::new().with_state("open"))?;

    rec.set_time_sequence("step", 1);
    rec.log("door", &rerun::StateChange::new().with_state("closed"))?;

    rec.set_time_sequence("step", 2);
    rec.log("door", &rerun::StateChange::new().with_state("open"))?;

    Ok(())
}
